azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -372,14 +376,43 @@ public void uploadProjectFile(Project project, int version, String filetype,
     }
   }
 
-  private void uploadProjectFile(Connection connection, Project project,
-      int version, String filetype, String filename, File localFile,
+  private void uploadProjectFile(Connection connection, int projectId, int version, File localFile)
```

Comment:
I don' think we do apart from that the outer api does connection handling, etc making the inner function simpler.